lib/models/shop/shop.dart

```diff
@@ -1,17 +1,43 @@
+import 'package:equatable/equatable.dart';
 import 'package:json_annotation/json_annotation.dart';
 import 'package:kaizen/models/common/reward/reward.dart';
 part 'shop.g.dart';
 
 @JsonSerializable()
-class Shop {
-  final Reward milestone;
+class Shop extends Equatable {
+  final Reward? milestone;
```

Comment:
La milestone in questo caso è un `Nullable Reward` perché altrimenti se un utente elimina dallo shop una Reward l'indice si sfalza e in questo modo non gestisco nemmeno sorting. MI sembrava soluzione migliore, se avete altre idee suggerite pure, ma non credo cambi troppo.

Reply:
Effettivamente non ci abbiamo pensato ieri. La milestone come Reward? per me è buona soluzione.